index.js

```diff
@@ -1505,6 +1505,29 @@
   }
   S.pipe = def('pipe', {}, [$.Array($.AnyFunction), a, b], pipe);
 
+  //# pipeK :: Chain m => [(a -> m b), ..., (y -> m z)] -> m a -> m z
```

Comment:
Since it's built on `Z.reduce`, maybe the type can be generalised.

```hs
pipeK :: (Foldable f, Chain m) => f (Any -> m Any) -> m a -> m b
```

Not sure what to do about the `Any` stuff, but it's probably better than using `AnyFunction`.

Reply:
Good idea, Aldwin. Perhaps one of us should open a pull request to generalize `S.pipe`.

Reply:
I had considered this, but had two reservations:
1. I thought generalizing `pipe` and `pipeK` belonged in a separate PR
2. I wasn't certain generalizing would be useful. If `sanctuary-list` existed it would be a different story.

Reply:
> I thought generalizing `pipe` and `pipeK` belonged in a separate PR

I agree. If we are to generalize, the neatest order of operations would be to generalize `pipe` in a separate pull request, merge that pull request, then rebase this pull request and generalize `pipeK`.

Reply:
#473

Reply:
The `pipe` pull request has been merged, so this pull request can now be updated. :)